Fluid/Ast/TextSpanStatement.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Parlot;
+using System.IO;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+
+namespace Fluid.Ast
+{
+    public class TextSpanStatement : Statement
+    {
+        private bool _isStripped = false;
+        private bool _isEmpty = false;
+        private readonly object _synLock = new object();
+        private TextSpan _text;
+
+        public TextSpanStatement(TextSpan text)
+        {
+            _text = text;
+        }
+
+        public TextSpanStatement(string text)
+        {
+            _text = new TextSpan(text);
+        }
+
+        public bool StrippedLeft { get; set; }
+        public bool StrippedRight { get; set; }
+
+        public void StripRight()
+        {
+            StrippedRight = true;
+        }
+
+        public void StripLeft()
+        {
+            StrippedLeft = true;
+        }
+
+        public TextSpan Text => _text;
+
+        public override ValueTask<Completion> WriteToAsync(TextWriter writer, TextEncoder encoder, TemplateContext context)
+        {
+            // Prevent this instance from mutating concurrently
+
+            if (!_isStripped)
+            {
+                lock (_synLock)
```

Comment:
can this lock be shorter? like calculating the end result and then finally taking the lock? don't know the contention amounts here